src/NServiceBus.Core/IdGeneration/CombGuid.cs

```diff
@@ -1,42 +1,60 @@
 namespace NServiceBus
 {
     using System;
+    using System.Buffers.Binary;
+#if NETFRAMEWORK
+    using System.Runtime.InteropServices;
+#endif
 
     /// <summary>
     /// Generates a Guid using http://www.informit.com/articles/article.asp?p=25862
     /// The Comb algorithm is designed to make the use of <see cref="Guid"/>s as Primary Keys, Foreign Keys, and Indexes nearly as efficient
     /// as <see cref="int"/>.
     /// </summary>
-    /// <remarks>Source: https://github.com/nhibernate/nhibernate-core/blob/4.0.4.GA/src/NHibernate/Id/GuidCombGenerator.cs</remarks>
+    /// <remarks>Original Source: https://github.com/nhibernate/nhibernate-core/blob/4.0.4.GA/src/NHibernate/Id/GuidCombGenerator.cs</remarks>
     static class CombGuid
     {
-        /// <summary>
-        /// Generate a new <see cref="Guid" /> using the comb algorithm.
-        /// </summary>
         public static Guid Generate()
         {
-            var guidArray = Guid.NewGuid().ToByteArray();
+            var newGuid = Guid.NewGuid();
+            Span<byte> guidArray = stackalloc byte[16];
+#if NETCOREAPP
+            if (!newGuid.TryWriteBytes(guidArray))
+#elif NETFRAMEWORK
+            if (!MemoryMarshal.TryWrite(guidArray, ref newGuid))
+#endif
+            {
+
+                guidArray = newGuid.ToByteArray();
+            }
 
             var now = DateTime.UtcNow; // Internal use, no need for DateTimeOffset
 
             // Get the days and milliseconds which will be used to build the byte string
             var days = new TimeSpan(now.Ticks - BaseDateTicks);
             var timeOfDay = now.TimeOfDay;
 
-            // Convert to a byte array
-            // Note that SQL Server is accurate to 1/300th of a millisecond so we divide by 3.333333
-            var daysArray = BitConverter.GetBytes(days.Days);
```

Comment:
> The order of bytes in the array returned by the GetBytes method depends on whether the computer architecture is little-endian or big-endian.

I realized I need to take this into account because I was assuming I can remove the Reverse when writing BigEndian. Now I'm thinking if the original code was always assuming a fixed endianness because it always reversed the order of bytes

Reply:
Pushed an update